code/src/ng_link/ng_state.py

```diff
@@ -0,0 +1,348 @@
+        
+        self.__state = self.get_state()
+    
+    def get_state(self) -> dict:
+        """
+        Property getter of state.
+        
+        Returns
+        ------------------------
+        dict
+            Dictionary with the actual layer state.
+        """
+        
+        actual_state = {}
+        actual_state['dimensions'] = {}
+
+        # Getting actual state for all attributes
+        for axis, value_list in self.dimensions.items():
+            actual_state['dimensions'][axis] = value_list
+            
+        actual_state['layers'] = self.__layers
+        
+        return actual_state
+    
+    def save_state_as_json(
+        self, 
+        output_json:Optional[PathLike]='', 
+        update_state:Optional[bool]=False
+    ) -> None:
+        """
+        Saves a neuroglancer state as json.
+        
+        Parameters
+        ------------------------
+        output_json: Optional[PathLike]
+            Path where the neuroglancer state will be written as json
+        
+        update_state: Optional[bool]
+            Updates the neuroglancer state with dimensions and layers in case they were changed 
+            using class methods.
+        """
+        output_json = str(output_json)
+        
+        if not len(output_json):
+            output_json = self.output_json
+        else:
+            output_json = self.__fix_output_json_path(output_json)
+            self.output_json = output_json
+        
+        if update_state:
+            self.__state = self.state()
+    
+        utils.save_dict_as_json(output_json, self.__state, verbose=self.verbose)
+    
+    def get_url_link(
+        self, 
+        base_url:Optional[str]='https://neuroglancer-demo.appspot.com/',
+        save_txt:Optional[bool]=True,
+        output_txt:Optional[PathLike]=''
+    ) -> str:
+        """
+        Creates the neuroglancer link based on where the json will be written.
+        
+        Parameters
+        ------------------------
+        base_url: Optional[str]
+            Base url where neuroglancer app was deployed. Default: https://neuroglancer-demo.appspot.com/
+        
+        save_txt: Optional[bool]
+            Saves the url link to visualize data as a .txt file in a specific path given by 
+            output_txt parameter.
+        
+        output_txt: Optional[PathLike]
+            Path where the .txt file will be written. Default: '' which then will be converted to 
+            self.output_json value.
+        
+        Returns
+        ------------------------
+        str
+            Neuroglancer url to visualize data.
+        """
+        output_txt = str(output_txt)
+        
+        json_path = str(self.output_json)
+        json_path = 'gs://' + json_path
+        
+        link = f"{base_url}#!{json_path}"
+        
+        if save_txt:
+            
+            if not len(output_txt):
+                output_txt = self.output_json.parent

```

Comment:
Would it be easier just to put the link directly in the output JSON, instead of the output JSON needing to refer to a separate text file?

Reply:
I can include the link in the json file. People will need to download the json from the bucket to copy and paste it in the navigator and it won't be visible in the json state that's shown in neuroglancer.

Reply:
That would be true for a text file too, right?